qiskit_experiments/library/characterization/fine_frequency.py

```diff
@@ -0,0 +1,129 @@
+
+        .. parsed-literal::
+
+                    ┌────┐┌────────────────┐┌──────────┐┌────┐ ░ ┌─┐
+                 q: ┤ √X ├┤ Delay(n*τ[dt]) ├┤ Rz(nπ/2) ├┤ √X ├─░─┤M├
+                    └────┘└────────────────┘└──────────┘└────┘ ░ └╥┘
+            meas: 1/══════════════════════════════════════════════╩═
+                                                                  0
+    # section: analysis_ref
+        :py:class:`FineFrequencyAnalysis`
+    """
+
+    def __init__(
+        self,
+        qubit: int,
+        delay_duration: int,
+        backend: Optional[Backend] = None,
+        repetitions: Optional[List[int]] = None,
+    ):
+        """Setup a fine frequency experiment on the given qubit.
+
+        Args:
+            qubit: The qubit on which to run the fine frequency characterization experiment.
+            delay_duration: The duration of the delay at :math:`n=1`.
+            backend: Optional, the backend to run the experiment on.
+            repetitions: The number of repetitions, if not given then the default value
+                from the experiment default options will be used.
+        """
+        super().__init__([qubit], analysis=FineFrequencyAnalysis(), backend=backend)
+
+        if repetitions is not None:
+            self.set_experiment_options(repetitions=repetitions)
+
+        self.set_experiment_options(delay_duration=delay_duration)
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        r"""Default values for the fine frequency experiment.
+
+        Experiment Options:
+            repetitions (List[int]): A list of the number of times that the identity is repeated.
+            delay_duration (int): The duration of the delay as the number of arbitrary waveform
+                generator samples it contains. The total length of the delay will be n times
+                ``delay_duration`` where n also determines the rotation angle of the ``RZGate``
+                by :math:`n \pi/2`.
+        """
+        options = super()._default_experiment_options()
+        options.repetitions = list(range(40))
+        options.delay_duration = None
+
+        return options
+
+    @classmethod
+    def _default_transpile_options(cls) -> Options:
+        """Default transpiler options."""
+        options = super()._default_transpile_options()
+        options.inst_map = None
+        return options
+
+    def _pre_circuit(self) -> QuantumCircuit:
+        """A method that subclasses can override to perform gates before the main sequence."""
+        return QuantumCircuit(1)
+
+    def circuits(self) -> List[QuantumCircuit]:
+        """Return the list of quantum circuits to run."""
+
+        circuits = []
+
+        # The main sequence
+        for repetition in self.experiment_options.repetitions:
+            circuit = self._pre_circuit()
+            circuit.sx(0)
+
+            circuit.delay(duration=self.experiment_options.delay_duration * repetition)
```

Comment:
```suggestion
            circuit.delay(0, duration=self.experiment_options.delay_duration * repetition , "dt")
```
you need to specify qubit, also unit to clarify

Reply:
Good catch thanks.

Reply:
Actually, no. The suggestion does not work as is. `qarg` is optional see here: https://github.com/Qiskit/qiskit-terra/blob/d9ab98f2b3663b42ba354ed1cd2bc18b3f2cd19d/qiskit/circuit/quantumcircuit.py#L2701-L2706 I think I will leave this empty but adding unit is a good idea for clarity. See here: f05e61f